tools/checkpoint_conversion/convert_xlm_roberta_checkpoints.py

```diff
@@ -0,0 +1,334 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+import numpy as np
+import tensorflow as tf
+import torch
+import transformers
+from absl import app
+from absl import flags
+from checkpoint_conversion_utils import get_md5_checksum
+from tensorflow import keras
+
+import keras_nlp
+
+PRESET_MAP = {
+    "xlm_roberta_base": ("xlmr.base", "xlm-roberta-base"),
+    "xlm_roberta_large": ("xlmr.large", "xlm-roberta-large"),
+}
+
+DOWNLOAD_SCRIPT_URL = (
+    "https://dl.fbaipublicfiles.com/fairseq/models/xlmr.{}.tar.gz"
+)
+
+EXTRACT_DIR = "./{}"
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "preset", None, f'Must be one of {",".join(PRESET_MAP.keys())}'
+)
+
+
+def download_model(preset, size):
+    print("-> Download original weights.")
+    archive_file_path = keras.utils.get_file(
+        fname=None,
+        origin=f"https://dl.fbaipublicfiles.com/fairseq/models/{size}.tar.gz",
+    )
```

Comment:
Should we add a `cache_subdir=os.join("checkpoint_conversion", preset)` here? Might be nice if all this scripts downloaded data to the same dir so we can delete them in one go.